### PR TITLE
server: use a checker controller in coordinator

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Currently, we need to initialize all checkers in `coordinator`, which is untidy.

### What is changed and how it works?
This PR makes some changes:

1. use a checker controller to initialize all checkers. Especially, for merge checker, it is not supported, we won't initialize it.

2. Move `Cluster` interface to `opt.go` temporarily to reduce the duplication of the interface.
3. Move `checker` package into `schedule` directory.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/coordinator.go [hunk 1]
```diff
@@ -52,38 +50,35 @@ var (
...
+	minSupportVersion := *MinSupportedVersion(RegionMerge)
+	isSupportMerge := !clusterVersion.LessThan(minSupportVersion)
 	return &coordinator{
-		ctx:              ctx,
-		cancel:           cancel,
-		cluster:          cluster,
-		learnerChecker:   checker.NewLearnerChecker(),
-		replicaChecker:   checker.NewReplicaChecker(cluster, classifier),
-		namespaceChecker: checker.NewNamespaceChecker(cluster, classifier),
-		mergeChecker:     checker.NewMergeChecker(cluster, classifier),
```
[the remark this answers is not in the document]
Comment (rleungx) [reply server/coordinator.go]: I think it should have a check before, but actually, we don't have.

**Comment by Luffbee** [comment]
@disksing has some different opinions on the `Cluster` interface problem, I think we need to discuss about it.

**Comment by rleungx** [comment]
@nolouch @disksing @lhy1024 PTAL.

File: server/cluster_worker.go [hunk 2]
```diff
@@ -120,13 +116,19 @@ func (c *RaftCluster) handleAskBatchSplit(request *pdpb.AskBatchSplitRequest) (*
...
 
-		c.coordinator.mergeChecker.RecordRegionSplit(newRegionID)
+		recordRegions = append(recordRegions, newRegionID)
 		splitIDs = append(splitIDs, &pdpb.SplitID{
 			NewRegionId: newRegionID,
 			NewPeerIds:  peerIDs,
 		})
 	}
 
+	recordRegions = append(recordRegions, reqRegion.GetId())
```
Comment (lhy1024) [inline server/cluster_worker.go]: replace `GetId` with `GetID`?

**Pull request merged**